core/src/main/java/com/graphhopper/routing/profiles/BitEncodedValue.java

```diff
@@ -0,0 +1,54 @@
+package com.graphhopper.routing.profiles;
+
+/**
+ * This class provides easy access to just one bit.
+ */
+public final class BitEncodedValue extends IntEncodedValue {
```

Comment:
I made the difference here because boolean implies 1 byte and double implies 8 bytes. For Integer I just didn't find a better word yet.